core/executor/src/wasm_executor.rs

```diff
@@ -767,6 +768,42 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 
 		Ok(offset)
 	},
+	ext_local_network_state(msg_len: *mut u32) -> *mut u8 => {
+		let res = this.ext.offchain()
+			.map(|api| api.local_network_state())
+			.ok_or_else(|| "Calling unavailable API ext_local_network_state: wasm")?;
+
+		let encoded = res.encode();
+		let len = encoded.len() as u32;
+		let offset = this.heap.allocate(len)? as u32;
+		this.memory.set(offset, &encoded)
+			.map_err(|_| "Invalid attempt to set memory in ext_local_network_state")?;
+
+		this.memory.write_primitive(msg_len, len)
```

Comment:
I think it would be better if we change the name of `msg_len` to something difference, since I find this line quite confusing. `written_out`, as used in the sibiling function, works better IMO.